tests/pyccel/scripts/generic_functions.py

```diff
@@ -79,6 +79,15 @@ def array_elem1(x):
 def multi_tmplt_2(y, z):
     return y + z
 
+@template('g', types=['int', 'int'])
```

Comment:
Does this only handle explicitly duplicated types or also duplicated types with different names (e.g. `float64` == `real`)?

Reply:
it handles duplicated types with different names as well, I will add some tests for this case

Reply:
Do we still support `real` in annotations?